### ISSUE TITLE
ProgrammingError: relation "device_lists_stream_user_id" already exists

### ISSUE THREAD

**Issue by chazdiezal** [matrix-org/synapse#2135]
log is filled with this error...

```
2017-04-19 16:18:10,661 - synapse.handlers.presence - 240 - INFO - - Performing                                              _persist_unpersisted_changes. Persisting 1 unpersisted changes
2017-04-19 16:18:10,747 - synapse.handlers.presence - 252 - INFO - - Finished _p                                             ersist_unpersisted_changes
2017-04-19 16:18:11,827 - synapse.storage.background_updates - 148 - INFO - - St                                             arting update batch on background update 'device_lists_stream_idx'
2017-04-19 16:18:11,847 - synapse.storage.background_updates - 266 - INFO - - Ad                                             ding index device_lists_stream_user_id to device_lists_stream
2017-04-19 16:18:11,867 - synapse.storage.background_updates - 102 - ERROR - - E                                             rror doing update
Traceback (most recent call last):
  File "/root/.synapse/local/lib/python2.7/site-packages/synapse/storage/backgro                                             und_updates.py", line 99, in start_doing_background_updates
    self.BACKGROUND_UPDATE_DURATION_MS
  File "/root/.synapse/local/lib/python2.7/site-packages/twisted/internet/defer.                                             py", line 1299, in _inlineCallbacks
    result = result.throwExceptionIntoGenerator(g)
  File "/root/.synapse/local/lib/python2.7/site-packages/twisted/python/failure.                                             py", line 393, in throwExceptionIntoGenerator
    return g.throw(self.type, self.value, self.tb)
  File "/root/.synapse/local/lib/python2.7/site-packages/synapse/storage/backgro                                             und_updates.py", line 142, in do_next_background_update
    res = yield self._do_background_update(update_name, desired_duration_ms)
  File "/root/.synapse/local/lib/python2.7/site-packages/twisted/internet/defer.                                             py", line 1299, in _inlineCallbacks
    result = result.throwExceptionIntoGenerator(g)
  File "/root/.synapse/local/lib/python2.7/site-packages/twisted/python/failure.                                             py", line 393, in throwExceptionIntoGenerator
    return g.throw(self.type, self.value, self.tb)
  File "/root/.synapse/local/lib/python2.7/site-packages/synapse/storage/backgro                                             und_updates.py", line 176, in _do_background_update
    items_updated = yield update_handler(progress, batch_size)
  File "/root/.synapse/local/lib/python2.7/site-packages/twisted/internet/defer.                                             py", line 1299, in _inlineCallbacks
    result = result.throwExceptionIntoGenerator(g)
  File "/root/.synapse/local/lib/python2.7/site-packages/twisted/python/failure.                                             py", line 393, in throwExceptionIntoGenerator
    return g.throw(self.type, self.value, self.tb)
  File "/root/.synapse/local/lib/python2.7/site-packages/synapse/storage/backgro                                             und_updates.py", line 268, in updater
    yield self.runWithConnection(create_index_concurrently)
  File "/root/.synapse/local/lib/python2.7/site-packages/twisted/internet/defer.                                             py", line 1299, in _inlineCallbacks
    result = result.throwExceptionIntoGenerator(g)
  File "/root/.synapse/local/lib/python2.7/site-packages/twisted/python/failure.                                             py", line 393, in throwExceptionIntoGenerator
    return g.throw(self.type, self.value, self.tb)
  File "/root/.synapse/local/lib/python2.7/site-packages/synapse/storage/_base.p                                             y", line 347, in runWithConnection
    inner_func, *args, **kwargs
  File "/root/.synapse/local/lib/python2.7/site-packages/twisted/python/threadpo                                             ol.py", line 250, in inContext
    result = inContext.theWork()
  File "/root/.synapse/local/lib/python2.7/site-packages/twisted/python/threadpo                                             ol.py", line 266, in <lambda>
    inContext.theWork = lambda: context.call(ctx, func, *args, **kw)
  File "/root/.synapse/local/lib/python2.7/site-packages/twisted/python/context.                                             py", line 122, in callWithContext
    return self.currentContext().callWithContext(ctx, func, *args, **kw)
  File "/root/.synapse/local/lib/python2.7/site-packages/twisted/python/context.                                             py", line 85, in callWithContext
    return func(*args,**kw)
  File "/root/.synapse/local/lib/python2.7/site-packages/twisted/enterprise/adba                                             pi.py", line 307, in _runWithConnection
    compat.reraise(excValue, excTraceback)
  File "/root/.synapse/local/lib/python2.7/site-packages/twisted/enterprise/adba                                             pi.py", line 298, in _runWithConnection
    result = func(conn, *args, **kw)
  File "/root/.synapse/local/lib/python2.7/site-packages/synapse/storage/_base.p                                             y", line 343, in inner_func
    return func(conn, *args, **kwargs)
  File "/root/.synapse/local/lib/python2.7/site-packages/synapse/storage/backgro                                             und_updates.py", line 257, in create_index_concurrently
    c.execute(sql)
```

**Comment by richvdh** [comment]
shrike is seeing this too... we should probably figure out what is causing it

**Comment by richvdh** [comment]
Fixed by #2167
